### PR TITLE
AADConditionalAccessPolicy include-excludeGuestsorExternalUsers and TransferMethods

### LINKED ISSUE #4472: AADConditionalAccessPolicy: Add support for Authentication Flows
### Description of the issue

`AADConditionalAccessPolicy` does not currently support Conditional Access (CA) policy settings related to Authentication flows:

- Device Code Flow
- Authentication transfer

See attached photo for corresponding portal based settings:

image

While this CA policy can be successfully created in the Portal, an export of the policy does not reflect any of the settings related to Authentication flow settings.

See attached export of the resource for details.

Enhancement request:
Add support to the `AADConditionalAccessPolicy` resource for adjusting the `Device Code Flow` and `Authentication transfer` settings.

### Microsoft 365 DSC Version

v1.24.313.1

### Which workloads are affected

Azure Active Directory

### The DSC configuration

```powershell
AADConditionalAccessPolicy "AADConditionalAccessPolicy-Block MS Device code flow"
{
    ApplicationId                        = $ConfigurationData.NonNodeData.ApplicationId;
    AuthenticationContexts               = @();
    BuiltInControls                      = @("block");
    CertificateThumbprint                = $ConfigurationData.NonNodeData.CertificateThumbprint;
    ClientAppTypes                       = @("all");
    CloudAppSecurityType                 = "";
    CustomAuthenticationFactors          = @();
    DeviceFilterRule                     = "";
    DisplayName                          = "Block MS Device code flow";
    Ensure                               = "Present";
    ExcludeApplications                  = @();
    ExcludeExternalTenantsMembers        = @();
    ExcludeExternalTenantsMembershipKind = "";
    ExcludeGroups                        = @();
    ExcludeLocations                     = @();
    ExcludePlatforms                     = @();
    ExcludeRoles                         = @();
    ExcludeUsers                         = @("");
    GrantControlOperator                 = "OR";
    Id                                   = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx";
    IncludeApplications                  = @("All");
    IncludeExternalTenantsMembers        = @();
    IncludeExternalTenantsMembershipKind = "";
    IncludeGroups                        = @();
    IncludeLocations                     = @();
    IncludePlatforms                     = @();
    IncludeRoles                         = @();
    IncludeUserActions                   = @();
    IncludeUsers                         = @("All");
    PersistentBrowserMode                = "";
    SignInFrequencyType                  = "";
    SignInRiskLevels                     = @();
    State                                = "enabled";
    TenantId                             = $OrganizationName;
    UserRiskLevels                       = @();
}
```

### Verbose logs showing the problem

```powershell
N/A
```

### Environment Information + PowerShell Version

_No response_

### LINKED ISSUE #4630: AADConditionalAccessPolicy: ExcludeGuestOrExternalUserTypes with value "b2bCollaborationGuest" is not applied
### Description of the issue

I want to create a conditional access policy who use the parameter "Guest or external users" in the exclude part. When I extracted my conditional access policy I get the following value :
ExcludeGuestOrExternalUserTypes      = @("b2bCollaborationGuest");
But when I use it in my deployment the value is not applied and guest account are not excluded.
I have the problem with the AuthenticationStrength parameter who is also not applied when set... I already opened another case for that.

### Microsoft 365 DSC Version

1.24.424.1

### Which workloads are affected

Azure Active Directory (Entra ID)

### The DSC configuration

```powershell
AADConditionalAccessPolicy "AADConditionalAccessPolicy-Require MFA and password change for high risk users"
        {
            AuthenticationContexts               = @();
            AuthenticationStrength               = "Multifactor authentication";
            BuiltInControls                      = @();
            ClientAppTypes                       = @("all");
            CloudAppSecurityType                 = "";
            Credential                           = $Credscredential;
            CustomAuthenticationFactors          = @();
            DeviceFilterRule                     = "";
            DisplayName                          = "Require MFA and password change for high risk users";
            Ensure                               = "Present";
            ExcludeApplications                  = @();
            ExcludeExternalTenantsMembers        = @();
            ExcludeExternalTenantsMembershipKind = "all";
            ExcludeGroups                        = @("GPAZ-AzureAD-MFA-Bypass");
            ExcludeGuestOrExternalUserTypes      = @("b2bCollaborationGuest");
            ExcludeLocations                     = @();
            ExcludePlatforms                     = @();
            ExcludeRoles                         = @();
            ExcludeUsers                         = @();
            GrantControlOperator                 = "AND";
            Id                                   = "";
            IncludeApplications                  = @("All");
            IncludeExternalTenantsMembers        = @();
            IncludeExternalTenantsMembershipKind = "";
            IncludeGroups                        = @();
            IncludeLocations                     = @();
            IncludePlatforms                     = @();
            IncludeRoles                         = @();
            IncludeUserActions                   = @();
            IncludeUsers                         = @("All");
            PersistentBrowserMode                = "";
            SignInFrequencyInterval              = "everyTime";
            SignInFrequencyIsEnabled             = $True;
            SignInFrequencyType                  = "";
            SignInRiskLevels                     = @();
            State                                = "enabled";
            UserRiskLevels                       = @("high");
        }
```

### Verbose logs showing the problem

_No response_

### Environment Information + PowerShell Version

_No response_

### PR DESCRIPTION
**Pull request by mibarm**
#### Pull Request (PR) description
excludeGuestsOrExternalUsers and includeGuestsOrExternalUsers cannot be set to null with
Graph PowerShell SDK but only with Invoke-MgGraphRequest. 
Also included AuthenticanFlow TransferMethods

#### This Pull Request (PR) fixes the following issues
    - Fixes #4630
    - Fixes #4472

### CODE REVIEW

**Review by NikCharlebois — APPROVED** [review]

**Comment by NikCharlebois** [comment]
Can you please merge the conflicts. Thanks

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADConditionalAccessPolicy/MSFT_AADConditionalAccessPolicy.schema.mof [hunk 1]
```diff
@@ -44,6 +44,7 @@ class MSFT_AADConditionalAccessPolicy : OMI_BaseResource
     [Write, Description("Specifies, whether Browser Persistence is controlled by the Policy.")] Boolean PersistentBrowserIsEnabled;
     [Write, Description("Specifies, what Browser Persistence control is enforced by the Policy."), ValueMap{"Always","Never",""}, Values{"Always","Never",""}] String PersistentBrowserMode;
     [Write, Description("Name of the associated authentication strength policy.")] String AuthenticationStrength;
+    [Write, Description("Names of the associated authentication flow transfer methods")] String Transfermethods;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADConditionalAccessPolicy/MSFT_AADConditionalAccessPolicy.schema.mof:47]: Picky, but can you make the M in Methods capital (property name). Thanks

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADConditionalAccessPolicy/MSFT_AADConditionalAccessPolicy.schema.mof (hunk 1 above)
Comment (mibarm) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADConditionalAccessPolicy/MSFT_AADConditionalAccessPolicy.schema.mof]: done :)

**Comment by mibarm** [comment]
@NikCharlebois anything left on my side that is still open so that the PR can be completed? thx

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**